src/Fmt_ast.ml

```diff
@@ -1395,7 +1395,10 @@ and fmt_constructor_arguments_result c ctx args res =
   let pre : _ format = if Option.is_none res then " of@ " else ": " in
   fmt_constructor_arguments c ctx pre args
   $ opt res (fun typ ->
-        fmt " " $ fmt "-> " $ fmt_core_type c (sub_typ ~ctx typ) )
+        let before_type : _ format =
+          match args with Pcstr_tuple [] -> ": " | _ -> "-> "
+        in
+        fmt " " $ fmt before_type $ fmt_core_type c (sub_typ ~ctx typ) )
```

Comment:
Maybe move this to before `let pre` to keep the formatting combinator code clearer?